snap/validate.go

```diff
@@ -71,6 +73,92 @@ func ValidateAlias(alias string) error {
 	return nil
 }
 
+// ValidateAppSocketName checks if a string ca be used as a name for a socket
+// (for socket activation).
+func ValidateAppSocketName(name string) error {
```

Comment:
why is this exported?

Reply:
I noticed most Validate* functions are exported. Also, it makes it possible to unittest the function in isolation

Reply:
I don't think it should be exported unless it's expected that people would call this from outside of this package. For testing, you make private things exported via `export_test.go`